lib/search.rb

```diff
@@ -265,6 +265,24 @@ def self.advanced_filters
     @advanced_filters
   end
 
+  advanced_filter(/^in:personal-direct$/) do |posts|
+    if @guardian.user
+      posts
+        .joins("JOIN topic_allowed_users tu ON posts.topic_id = tu.topic_id")
+        .joins("LEFT JOIN topic_allowed_groups tg ON posts.topic_id = tg.topic_id")
+        .where(<<~SQL, user_id: @guardian.user.id)
+          tg.id IS NULL
+          AND tu.user_id = :user_id
+          AND posts.topic_id IN (
+            SELECT topic_id
+            FROM topic_allowed_users
```

Comment:
For performance reasons you'll want to add the user and topics filters in this correlated subquery. Otherwise it will query all rows in the database!

Reply:
I've updated the PR and I think the query should be more efficient now. Let me know if we can do better. Thanks for the review!